File: app/src/main/java/com/mona/moviesapp/popular_people/PopularDetailsActivity.java
package com.mona.moviesapp.popular_people;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.widget.GridView;
import android.widget.ImageView;
import android.widget.TextView;

import com.mona.moviesapp.R;
import com.mona.moviesapp.popular_people.pojo.PopularInfo;
import com.mona.moviesapp.popular_people.pojo.PopularProfile;
import com.mona.moviesapp.popular_people.pojo.Profiles;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;
import java.util.concurrent.ExecutionException;

public class PopularDetailsActivity extends AppCompatActivity {

    private TextView name, depart, gender, popularity, adult;
    private ImageView profile;
    private RecyclerView gridphotos;

    private String popname, popdepart, popprofile;
    private Boolean popadult;
    private int popgender, popid;
    private float poppopular;

    URL ImageUrl = null;
    InputStream inputStream = null;
    Bitmap bmImg = null;

    GridLayoutManager gridLayoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_popular_details);

        name = findViewById(R.id.nametxt);
        depart = findViewById(R.id.departtxt);
        gender = findViewById(R.id.gendertxt);
        popularity = findViewById(R.id.popularitytxt);
        adult = findViewById(R.id.adulttxt);
        profile = findViewById(R.id.profileimg);
        gridphotos = findViewById(R.id.photos);

        Intent intent = getIntent();
        Bundle bundle =  intent.getBundleExtra("data");

        if(!bundle.isEmpty()){
            popname = bundle.getString("popName");
            popdepart = bundle.getString("popeDepart");
            popadult = bundle.getBoolean("popAdult");
            popgender = bundle.getInt("popGender");
            poppopular = bundle.getFloat("popPopular");
            popprofile = bundle.getString("profile");
            popid = bundle.getInt("id");
        }
        String ad = Boolean.toString(popadult);
        String pop = Float.toString(poppopular);
        String gend = Float.toString(popgender);
        name.setText(popname);
        depart.setText(popdepart);
        adult.setText("Adult: "+ad);
        gender.setText("Gender: "+gend);
        popularity.setText("Popularity: "+pop);

        try {
            profile.setImageBitmap(new downloadImage(profile).execute(popprofile).get());
        } catch (ExecutionException e) {
            e.printStackTrace();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        new popularPhotos().execute("https://api.themoviedb.org/3/person/"+popid+"/images?api_key=bd9eb9f62e484b7b3de4718afb6cd421");
    }

    public class downloadImage extends AsyncTask<String, Void, Bitmap> {

        public downloadImage(ImageView imageView) {

        }

        @Override
        protected Bitmap doInBackground(String... strings) {

            try {
                ImageUrl = new URL(strings[0]);
                Log.i("URL",ImageUrl.toString());

                HttpURLConnection conn = (HttpURLConnection) ImageUrl.openConnection();
                conn.setDoInput(true);
                conn.connect();
                inputStream = conn.getInputStream();
                bmImg = BitmapFactory.decodeStream(inputStream);

            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return bmImg;
        }

        @Override
        protected void onPostExecute(Bitmap bitmap) {
            super.onPostExecute(bitmap);
            if(bitmap != null){
                profile.setImageBitmap(bitmap);
            }
            else{
                profile.setImageResource(R.drawable.ic_launcher_background);
            }
        }
    }

    public class popularPhotos extends AsyncTask<String, String, String> {

        @Override
        protected String doInBackground(String... urls) {

            HttpURLConnection httpURLConnection = null;
            BufferedReader reader = null;

            try {
                URL url = new URL(urls[0]);
                httpURLConnection = (HttpURLConnection) url.openConnection();
                httpURLConnection.connect();

                InputStream stream = httpURLConnection.getInputStream();
                reader = new BufferedReader(new InputStreamReader(stream));
                StringBuffer buffer = new StringBuffer();

                String line = "";
                while ((line = reader.readLine()) != null) {
                    buffer.append(line);
                }
                return buffer.toString();
            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
            finally {
                if (httpURLConnection != null)
                    httpURLConnection.disconnect();
                try {
                    if (reader != null)
                        reader.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return null;
        }

        @Override
        protected void onPostExecute(String result) {
            super.onPostExecute(result);

            ArrayList<Profiles> profiles = new ArrayList<>();
            try {
                JSONObject popular_profiles = new JSONObject(result);
                JSONArray pop_profiles = popular_profiles.getJSONArray("profiles");
                for (int i = 0; i< pop_profiles.length(); i++) {
                    JSONObject profileResult = pop_profiles.getJSONObject(i);
                    Profiles pictures = new Profiles();
                    pictures.setFile_path(profileResult.getString("file_path"));
                    profiles.add(pictures);
                }

                ImagsAdapter adapter = new ImagsAdapter(profiles, PopularDetailsActivity.this);
                gridphotos.setAdapter(adapter);
                gridphotos.setLayoutManager(new GridLayoutManager(PopularDetailsActivity.this, 2));

            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
    }
}

File: app/src/main/java/com/mona/movies_app/popular_people/popular_details_screen/view/PopularDetailsActivity.kt
package com.mona.movies_app.popular_people.popular_details_screen.view

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide

import com.mona.movies_app.R
import com.mona.movies_app.popular_people.full_image_screen.view.FullImageActivity
import com.mona.movies_app.popular_people.popular_details_screen.interfaces.DetailsViewInterface
import com.mona.movies_app.popular_people.popular_details_screen.model.DetailsModel
import com.mona.movies_app.popular_people.popular_details_screen.presenter.DetailsPresenter
import com.mona.movies_app.popular_people.pojo.Profiles

import java.util.ArrayList

class PopularDetailsActivity : AppCompatActivity(), DetailsViewInterface {

    private var name: TextView? = null
    private var depart: TextView? = null
    private var gender: TextView? = null
    private var popularity: TextView? = null
    private var adult: TextView? = null
    private var profile: ImageView? = null
    private var gridPhotos: RecyclerView? = null
    private lateinit var detailsAdapter: ImagesAdapter
    private var gridLayoutManager: GridLayoutManager? = null

    var detailsPresenter: DetailsPresenter? = null
    private var popularPictures = ArrayList<Profiles>()

    private var detailsIntent: Intent? = null
    private var bundle: Bundle? = null

    private var popName: String? = null
    private var popDepart: String? = null
    private var popProfile: String? = null
    private var popAdult: Boolean? = null
    private var popGender: Int = 0
    private var popId: Int = 0
    private var popPopular: Float = 0.toFloat()

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_popular_details)

        name = findViewById(R.id.txt_name)
        depart = findViewById(R.id.txt_depart)
        gender = findViewById(R.id.txt_gender)
        popularity = findViewById(R.id.txt_popularity)
        adult = findViewById(R.id.txt_adult)
        profile = findViewById(R.id.profile_img)
        gridPhotos = findViewById(R.id.photos)

        gridLayoutManager = GridLayoutManager(this@PopularDetailsActivity, 2)

        configGridRecycleView(popularPictures)

        detailsPresenter = DetailsPresenter(this, DetailsModel())

        detailsIntent = intent
        bundle = detailsIntent!!.getBundleExtra("data")

        if (!bundle!!.isEmpty) {
            popName = bundle!!.getString("popName")
            popDepart = bundle!!.getString("popeDepart")
            popAdult = bundle!!.getBoolean("popAdult")
            popGender = bundle!!.getInt("popGender")
            popPopular = bundle!!.getFloat("popPopular")
            popProfile = bundle!!.getString("profile")
            popId = bundle!!.getInt("id")
        }

        name!!.text = popName
        depart!!.text = popDepart
        adult!!.text = "Adult: " + popAdult!!
        gender!!.text = "Gender: $popGender"
        popularity!!.text = "Popularity: $popPopular"

        Glide.with(this)
                .load(popProfile)
                .into(profile!!)

        detailsPresenter!!.setPopId(popId)

        profile!!.setOnClickListener {
            val intent = Intent(this@PopularDetailsActivity, FullImageActivity::class.java)
            val arg = Bundle()
            arg.putString("picture_path", popProfile)
            intent.putExtra("data", arg)
            startActivity(intent)
        }
    }

    override fun configGridRecycleView(profiles: ArrayList<Profiles>) {
        detailsAdapter = ImagesAdapter(profiles, this@PopularDetailsActivity)
        gridPhotos!!.adapter = detailsAdapter
        gridPhotos!!.setHasFixedSize(true)
        gridPhotos!!.setItemViewCacheSize(20)
        gridPhotos!!.isDrawingCacheEnabled = true
        gridPhotos!!.layoutManager = gridLayoutManager
    }

    override fun changeGrid() {
        gridPhotos!!.setHasFixedSize(true)
        gridPhotos!!.setItemViewCacheSize(20)
        gridPhotos!!.isDrawingCacheEnabled = true
        detailsAdapter.notifyDataSetChanged()
        gridPhotos!!.layoutManager = gridLayoutManager
    }

    override fun addPopularDetails(profiles: Profiles) {
        popularPictures.add(profiles)
    }

    override fun goToFullImageScreen(imgPath: String, prof: Profiles) {
        val intent = Intent(this, FullImageActivity::class.java)
        val arg = Bundle()
        arg.putString("picture_path", imgPath + prof.file_path)
        intent.putExtra("data", arg)
        startActivity(intent)
    }
}
